### ISSUE TITLE
BUG: CC0022 DisposableVariableNotDisposedAnalyzer: False positive in iterator methods

### ISSUE THREAD

**Issue by oysteinkrog** [code-cracker/code-cracker#877]
#Bug

This should not trigger CC0022:

````csharp
public IEnumerable<IDisposable> Foo()
{
    var disposable = new MemoryStream();
    yield return disposable;
}
````

This works:
````csharp
public IDisposable Foo()
{
    var disposable = new MemoryStream();
    return disposable;
}
````

**Comment by carloscds** [comment]
@oysteinkrog Bug confirmed!

**Comment by giggio** [comment]
Closed on #900
